app/mailers/reservation_mailer.rb

```diff
@@ -0,0 +1,7 @@
+class ReservationMailer < ApplicationMailer
+  def car_reservation_created
+    @reservation = params[:reservation]
+    @reserved_by_name = User.find(@reservation.reserved_by).display_name
+    mail(to: @reservation.program.unit.unit_preferences.find_by(name: "notification_email").value, subject: "New reservation for program: #{@reservation.program.title}" )
```

Comment:
_@reservation.program.unit.unit_preferences.find_by(name: "notification_email").value_

There is a chain of function calls happening in the above code. Is there a possibility that any one of them might return null or blank value ? Do we need any additional validation checks ?

Reply:
Good point. I will add default values in the parent template to be sure there is not a nil value.

Reply:
defaults have been added

Reply:
@rsmoke if @reservation.program.unit.unit_preferences.find_by(name: "notification_email").value is nil or empty mail command fails with "SMTP To address may not be blank". How the mailer can use the default to: value?